src/core/time.d

```diff
@@ -2282,52 +2620,155 @@ assert(before + timeElapsed == after).
+                _ticksPerSecond[i] = ticksPerSecond;
         }
-        else version(OSX)
+        else
+            assert(0, "Failed to get the frequency of the system's monotonic clock.");
+    }
+    else version(OSX)
+    {
+        mach_timebase_info_data_t info;
+        if(mach_timebase_info(&info) == 0)
         {
-            mach_timebase_info_data_t info;
-            if(mach_timebase_info(&info) == 0)
-                _ticksPerSecond = (1_000_000_000L * info.denom) / info.numer;
+            long ticksPerSecond = 1_000_000_000L * info.numer / info.denom;
+            foreach(i, typeStr; __traits(allMembers, ClockType))
+                _ticksPerSecond[i] = ticksPerSecond;
         }
-        else version(Posix)
+        else
+            assert(0, "Failed to get the frequency of the system's monotonic clock.");
+    }
+    else version(Posix)
+    {
+        timespec ts;
+        foreach(i, typeStr; __traits(allMembers, ClockType))
         {
-            timespec ts;
-            if(clock_getres(CLOCK_MONOTONIC, &ts) == 0)
+            mixin("enum clockType = ClockType." ~ typeStr ~ ";");
+            static if(clockType != ClockType.second)
             {
-                // For some reason, on some systems, clock_getres returns
-                // a resolution which is clearly wrong (it's a millisecond
-                // or worse, but the time is updated much more frequently
-                // than that). In such cases, we'll just use nanosecond
-                // resolution.
-                _ticksPerSecond = ts.tv_nsec >= 1000 ? 1_000_000_000L
-                                                     : 1_000_000_000L / ts.tv_nsec;
+                // This needs to be kept in sync with the corresponding code at the
+                // top of MonoTimeImpl, since as long as we're forced to put the
+                // static constructor outside of MonoTimeImpl, we don't have access
+                // to clockArg.
+                version(linux)
+                {
+                    import core.sys.linux.time;
+                    static if(clockType == ClockType.bootTime)            alias clockArg = CLOCK_BOOTTIME;
+                    else static if(clockType == ClockType.coarse)         alias clockArg = CLOCK_MONOTONIC_COARSE;
+                    else static if(clockType == ClockType.normal)         alias clockArg = CLOCK_MONOTONIC;
+                    else static if(clockType == ClockType.precise)        alias clockArg = CLOCK_MONOTONIC;
+                    else static if(clockType == ClockType.processCPUTime) alias clockArg = CLOCK_PROCESS_CPUTIME_ID;
+                    else static if(clockType == ClockType.raw)            alias clockArg = CLOCK_MONOTONIC_RAW;
+                    else static if(clockType == ClockType.threadCPUTime)  alias clockArg = CLOCK_THREAD_CPUTIME_ID;
+                    else static assert(0, "ClockType." ~ typeStr ~ " is not supported by MonoTimeImpl on this system.");
```

Comment:
This breaks the documentation build because with `version (CoreDdoc)` the enum contains all members.
I'd suggest to initialize the clock resolutions lazily instead.